src/parser/Parser.ts

```diff
@@ -184,1368 +188,1390 @@ export class Parser {
-                let right = expression();
-
-                // Create a dotted or indexed "set" based on the left-hand side's type
-                if (left instanceof Expr.IndexedGet) {
-                    consume(
-                        "Expected newline or ':' after indexed 'set' statement",
-                        Lexeme.Newline,
-                        Lexeme.Colon,
-                        Lexeme.Eof
-                    );
-
-                    return new Stmt.IndexedSet(
-                        left.obj,
-                        left.index,
-                        operator.kind === Lexeme.Equal
-                            ? right
-                            : new Expr.Binary(left, operator, right),
-                        left.closingSquare
-                    );
-                } else if (left instanceof Expr.DottedGet) {
-                    consume(
-                        "Expected newline or ':' after dotted 'set' statement",
-                        Lexeme.Newline,
-                        Lexeme.Colon,
-                        Lexeme.Eof
-                    );
+            while (this.match(Lexeme.Newline, Lexeme.Colon));
 
-                    return new Stmt.DottedSet(
-                        left.obj,
-                        left.name,
-                        operator.kind === Lexeme.Equal
-                            ? right
-                            : new Expr.Binary(left, operator, right)
-                    );
-                } else {
-                    return _expressionStatement();
-                }
-            } else {
-                return _expressionStatement();
-            }
+            return new Stmt.Increment(expr, operator);
         }
 
-        function printStatement(...additionalterminators: BlockTerminator[]): Stmt.Print {
-            let printKeyword = advance();
-
-            let values: (
-                | Expr.Expression
-                | Stmt.PrintSeparator.Tab
-                | Stmt.PrintSeparator.Space)[] = [];
-
-            //print statements can be empty, so look for empty print conditions
-            if (isAtEnd() || check(Lexeme.Newline, Lexeme.Colon)) {
-                let emptyStringLiteral = new Expr.Literal(new BrsString(""), printKeyword.location);
-                values.push(emptyStringLiteral);
-            } else {
-                values.push(expression());
-            }
+        if (!this.check(...additionalTerminators)) {
+            this.consume(
+                "Expected newline or ':' after expression statement",
+                Lexeme.Newline,
+                Lexeme.Colon,
+                Lexeme.Eof
+            );
+        }
 
-            while (!check(Lexeme.Newline, Lexeme.Colon, ...additionalterminators) && !isAtEnd()) {
-                if (check(Lexeme.Semicolon)) {
-                    values.push(advance() as Stmt.PrintSeparator.Space);
-                }
+        if (expr instanceof Expr.Call) {
+            re
```

Comment:
Renamed to `primaryExpression` to make it a little more clear what it does. This can be reverted if you don't like that.